components/board/genericlinux/gpio.go

```diff
@@ -312,11 +317,18 @@ func (pin *gpioPin) halfPwmCycle(ctx context.Context, shouldBeOn bool) bool {
 
 func (pin *gpioPin) softwarePwmLoop(ctx context.Context) {
 	for {
-		if !pin.halfPwmCycle(ctx, true) {
+		select {
+		case <-ctx.Done():
 			return
+		case <-pin.startSoftwarePWMChan:
```

Comment:
just want to make sure I understand: when the channel gets closed it goes into this case?

Reply:
Yes. If a channel gets closed and you attempt to read something out of it, you always get back a `nil` immediately.